pkg/applications/helmclient/client.go

```diff
@@ -0,0 +1,311 @@
+		Getters:          h.getterProviders,
+		RepositoryConfig: h.settings.RepositoryConfig,
+		RepositoryCache:  h.settings.RepositoryCache,
+		RegistryClient:   regClient,
+		Debug:            false,
+		Verify:           downloader.VerifyNever,
+		SkipUpdate:       true,
+		// todo credentials
+	}
+
+	chartToInstall, err := loader.Load(chartLoc)
+	if err != nil {
+		return nil, fmt.Errorf("can not load char: %w", err)
+	}
+
+	fi, err := os.Stat(chartLoc)
+	if err != nil {
+		return nil, fmt.Errorf("can not find chart at `%s': %w", chartLoc, err)
+	}
+
+	// If we got the chart from the filesystem (ie cloned from a git repository). We have to do build dependencies because
+	// charts directory may not exist.
+	//
+	// note: if we got the chart from a remote helm repository, we don't have to build dependencies because the package
+	// (ie the tgz) should already contain it.
+	if fi.IsDir() {
+		// Helm does not download dependency if the repository is unknown (ie not present in repository.xml)
+		// so we explicitly add to the repository file.
+		var dependencies []*chart.Dependency
+		if chartToInstall.Lock != nil {
+			dependencies = chartToInstall.Lock.Dependencies
+		} else {
+			dependencies = chartToInstall.Metadata.Dependencies
+		}
+
+		for _, dep := range dependencies {
+			// oci or file dependencies can not be added as a repository.
+			if strings.HasPrefix(dep.Repository, "http://") || strings.HasPrefix(dep.Repository, "https://") {
+				if _, err := h.ensureRepository(dep.Repository); err != nil {
+					return nil, fmt.Errorf("can not download index for repository: %w", err)
+				}
+			}
+		}
+
+		// Equivalent of helm dependency build.
+		err = man.Build()
+		if err != nil {
+			h.logger.Errorf("can not build dependencies for chart '%s'. log='%s'", chartLoc, out.String())
+			return nil, fmt.Errorf("can not build dependencies: %w", err)
+		}
+	}
+	h.logger.Debugf("succesully build dependencies for chart '%s'. log='%s'", chartLoc, out.String())
+	return chartToInstall, nil
+}
+
+// ensureRepository add the repository url if it's not already exist and download the last index file.
+// The repository is added with the name helm-manager-$(sha256 url).
+func (h HelmClient) ensureRepository(url string) (string, error) {
+	repoFile, err := repo.LoadFile(h.settings.RepositoryConfig)
+	if err != nil && !os.IsNotExist(errors.Cause(err)) {
+		return "", err
+	}
+
+	repoName, err := computeRepoName(url)
+	if err != nil {
+		return "", fmt.Errorf("can not compute repository's name for '%s': %w", url, err)
+	}
+	desiredEntry := &repo.Entry{
+		Name: repoName,
+		URL:  url,
+		// todo credential  ???
+	}
+
+	// Ensure we have the last version of the index file.
+	chartRepo, err := repo.NewChartRepository(desiredEntry, h.getterProviders)
+	if err != nil {
+		return "", err
+	}
+	// Constructor of the ChartRepository uses the default Helm cache. To avoid polluting the local repository when executing
+	// TU, we override it here.
```

Comment:
oops, my bad it's the french abbreviation for Unit Test. But since we don't use locking, we have to override the cache repository location anyway... 

I have updated the comment.